pkg/admission/init.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"reflect"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+type PluginInitializer interface {
+	Initialize(plugins []Interface)
```

Comment:
You mean sth like this? 
`type SharedInformersMap interface {
	GetSharedInformer (obj interface{})
}

type sharedInformersMap struct {
	sharedInformers map [reflect.Type] framework.SharedIndexInformer
}

func NewSharedInformersmap(client clientset.Interface) SharedInformersMap {
	infmap := &sharedInformersMap{
		sharedInformers : make(map[reflect.Type]framework.SharedIndexInformer),
	}
	infmap.sharedInformers[reflect.TypeOf(&api.Namespace{})] = informers.CreateSharedNamespaceIndexInformer(client, time.Minute*5)
	return infmap

}
func (i *sharedInformersMap)GetSharedInformer (obj interface{}) (framework.SharedIndexInformer, error) {
	if obj == nil {

	}
	v, ok := i.sharedInformers[reflect.TypeOf(obj)]
	if ok == true {
		nil, fmt.Errorf("Resources are not properly initialized: Missing informer for type %t", obj)
	}
	return v, nil
}`